src/net/rpc.rs

```diff
@@ -1509,6 +1510,86 @@ impl ConversationHttp {
         return response.send(http, fd).map(|_| ());
     }
 
+    /// Handle a GET health.
+    /// If the node's height is not up to the max height of all of its bootstrap nodes, we return
+    /// false for the health check.
+    /// Additionally return what percent of the max height the node height is at.
+    fn handle_get_health<W: Write>(
+        http: &mut StacksHttp,
+        fd: &mut W,
+        req: &HttpRequestType,
+        network_id: u32,
+        sortdb: &SortitionDB,
+        chainstate: &StacksChainState,
+        peerdb: &PeerDB,
+        block_stats: Option<&HashMap<NeighborKey, NeighborBlockStats>>,
+    ) -> Result<(), net_error> {
+        let response_metadata = HttpResponseMetadata::from(req);
+
+        let block_height = match chainstate.get_stacks_chain_tip(sortdb)? {
+            Some(tip) => tip.height,
+            None => 0,
+        };
+
+        let initial_neighbors =
+            PeerDB::get_valid_initial_neighbors(peerdb.conn(), network_id, block_height).unwrap();
+
+        if initial_neighbors.len() == 0 {
+            // no bootstrap nodes found, unable to determine health.
+            let response = HttpResponseType::NotFound(
+                response_metadata.clone(),
+                format!("No viable bootstrap peers found, unable to determine health"),
+            );
+            return response.send(http, fd).map(|_| ());
+        }
+
+        // get max block height amongst bootstrap nodes
+        let response = if let Some(block_stats) = block_stats {
+            let mut max_height: u64 = 1;
+            let mut stats_obtained = false;
+            for neighbor in initial_neighbors.iter() {
+                let nk = &neighbor.addr;
+                match block_stats.get(nk) {
+                    Some(neighbor_block_stats) => {
+                        if neighbor_block_stats.status == NodeStatus::Online {
+                            let height = neighbor_block_stats.inv.get_block_height();
+                            max_height = max_height.max(height);
+                            stats_obtained = true;
+                        }
+                    }
+                    None => {}
+                }
+            }
+
+            if !stats_obtained {
+                let response = HttpResponseType::NotFound(
+                    response_metadata.clone(),
+                    "Couldn't obtain stats on any bootstrap peers, unable to determine health"
+                        .to_string(),
+                );
+                return response.send(http, fd).map(|_| ());
+            }
+
+            let data = if block_height >= (max_height - 1) {
```

Comment:
I would defer to @kantai or @lgalabru here, but `max_height - 1` may give us a flapping health output over time.

Reply:
> @wileyj @CharlieC3 Wondering what your thoughts are on the response type in `handle_get_health` (in `rpc.rs`). Would you prefer
> 
> 1. always getting a HttpResponseType of GetHealth, with a detailed error field
> 2. always want a 400 level error if either there is an issue querying for the health or if the node is not healthy
> 3. a 400 level error if there is an issue querying for health, and a GetHealth response otherwise (includes instances when query succeeds, but it is determined the node is not healthy)

if the node is indeed "not healthy" or still syncing - it would need to be non-200 status in order to prevent traffic from being sent to it by a load balancer health check.

Reply:
i wouldn't think we need an "error" message, the same json output would be good to have for all requests, with the only difference between a "healthy" and "not healthy" response being the HTTP status code (ex: `200` status for a healthy node, `503` for unhealthy. subsets of 5xx can be used for more detailed reasons if you prefer)

Reply:
> I would defer to @kantai or @lgalabru here, but `max_height - 1` may give us a flapping health output over time.

The thresholds we incorporate in the health probes can account for this to only take action after seeing failing health for a while, like 30 minutes for example (considering how long a node may take to download a block from neighbors in rare times). It may only look strange to others hitting this endpoint in their browser and seeing a failed response every so often.

If the gap is widened beyond `max_height - 1`, to `max_height - 3` for example, it will take us longer to start detecting issues with the node since blocks can sometimes take an hour to be produced.